lib/trino-hive-formats/src/main/java/io/trino/hive/formats/avro/NativeLogicalTypesAvroTypeManager.java

```diff
@@ -0,0 +1,341 @@
+                        TimeType.TIME_MILLIS.writeLong(builder, i.longValue() * Timestamps.PICOSECONDS_PER_MILLISECOND);
+                    };
+                }
+                throw new IllegalStateException("Unreachable unfiltered logical type");
+            }
+            case TIME_MICROS -> {
+                if (schema.getType() == Schema.Type.LONG) {
+                    yield (builder, obj) -> {
+                        Long i = (Long) obj;
+                        TimeType.TIME_MICROS.writeLong(builder, i * Timestamps.PICOSECONDS_PER_MICROSECOND);
+                    };
+                }
+                throw new IllegalStateException("Unreachable unfiltered logical type");
+            }
+            case UUID -> {
+                if (schema.getType() == Schema.Type.STRING) {
+                    yield (builder, obj) -> UuidType.UUID.writeSlice(builder, javaUuidToTrinoUuid(java.util.UUID.fromString(obj.toString())));
+                }
+                throw new IllegalStateException("Unreachable unfiltered logical type");
+            }
+            default -> throw new IllegalStateException("Unreachable unfiltered logical type");
+        };
+    }
+
+    private <T> Optional<T> validateAndProduceFromName(Schema schema, Function<LogicalType, T> produce)
+    {
+        // TODO replace with switch sealed class syntax when stable
+        ValidateLogicalTypeResult logicalTypeResult = validateLogicalType(schema);
+        if (logicalTypeResult instanceof NoLogicalType ignored) {
+            return Optional.empty();
+        }
+        if (logicalTypeResult instanceof NonNativeAvroLogicalType ignored) {
+            log.debug("Unrecognized logical type " + schema);
+            return Optional.empty();
+        }
+        if (logicalTypeResult instanceof InvalidNativeAvroLogicalType invalidNativeAvroLogicalType) {
+            log.debug(invalidNativeAvroLogicalType.getCause(), "Invalidly configured native avro logical type");
+            return Optional.empty();
+        }
+        if (logicalTypeResult instanceof ValidNativeAvroLogicalType validNativeAvroLogicalType) {
+            return Optional.of(produce.apply(validNativeAvroLogicalType.getLogicalType()));
+        }
+        throw new IllegalStateException("Unhandled validate logical type result");
+    }
+
+    protected static ValidateLogicalTypeResult validateLogicalType(Schema schema)
+    {
+        final String typeName = schema.getProp(LogicalType.LOGICAL_TYPE_PROP);
+        if (typeName == null) {
+            return new NoLogicalType();
+        }
+        LogicalType logicalType;
+        switch (typeName) {
+            case TIMESTAMP_MILLIS, TIMESTAMP_MICROS, DECIMAL, DATE, TIME_MILLIS, TIME_MICROS, UUID:
+                logicalType = fromSchemaIgnoreInvalid(schema);
+                break;
+            case LOCAL_TIMESTAMP_MICROS + LOCAL_TIMESTAMP_MILLIS:
+                log.debug("Logical type " + typeName + " not currently supported by by Trino");
```

Comment:
Is this something we plan to support? I'm wondering how the log message might be useful.

Reply:
I would actually like to talk with you about this. I want to make sure I understand our `TimestampWithTimeZoneType` enough to know if maps directly to this logical type or can be coerced in a way that makes sense.